contracts/protocols/GnosisAuction.sol

```diff
@@ -1,4 +1,114 @@
+        AuctionDetails memory auctionDetails
+    ) internal {
+        (uint96 optionPremium, uint96 oTokenSellAmount) =
+            setupAuctionParameters(
+                auctionDetails.oTokenAddress,
+                gnosisEasyAuction,
+                auctionDetails.underlying,
+                optionsPremiumPricer,
+                auctionDetails.premiumDiscount
+            );
+
+        IERC20(auctionDetails.oTokenAddress).safeApprove(gnosisEasyAuction, 0);
+        IERC20(auctionDetails.oTokenAddress).safeApprove(
+            gnosisEasyAuction,
+            IERC20(auctionDetails.oTokenAddress).balanceOf(address(this))
+        );
+
+        uint256 auctionCounter =
+            IGnosisAuction(gnosisEasyAuction).initiateAuction(
+                auctionDetails.oTokenAddress, // address of oToken we minted and are selling
+                auctionDetails.asset, // address of asset we want in exchange for oTokens. Should match vault collateral
+                block.timestamp.add(auctionDetails.duration.div(2)), // orders can be cancelled before the auction's halfway point
+                block.timestamp.add(auctionDetails.duration), // order will last for `duration`
+                oTokenSellAmount, // we are selling all of the otokens minus a fee taken by gnosis
+                optionPremium, // the minimum we are willing to sell the oTokens for. A discount is applied on black-scholes price
+                1, // the minimum bidding amount must be 1 * 10 ** -assetDecimals
+                0, // the min funding threshold
+                false, // no atomic closure
+                auctionDetails.manager, // manager of auction
+                bytes("") // bytes for storing info like a whitelist for who can bid
+            );
+
+        console.log(auctionDetails.oTokenAddress);
+        console.log(auctionDetails.asset);
+        console.log(oTokenSellAmount);
+        console.log(optionPremium);
+
+        emit InitiateGnosisAuction(
+            auctionDetails.oTokenAddress,
+            auctionDetails.asset,
+            auctionCounter,
+            auctionDetails.manager
+        );
+    }
+
+    function setupAuctionParameters(
+        address oTokenAddress,
+        address gnosisEasyAuction,
+        address underlying,
+        address optionsPremiumPricer,
+        uint256 premiumDiscount
+    ) internal returns (uint96 optionPremium, uint96 oTokenSellAmount) {
+        IOtoken newOToken = IOtoken(oTokenAddress);
+        IGnosisAuction auction = IGnosisAuction(gnosisEasyAuction);
+        // We take our current oToken balance and we subtract an
+        // amount that is the fee gnosis takes. That will be our sell amount
+        // but gnosis will transfer all the otokens
+        oTokenSellAmount = uint96(
+            IERC20(oTokenAddress)
+                .balanceOf(address(this))
+                .mul(auction.FEE_DENOMINATOR())
+                .div(auction.FEE_DENOMINATOR().add(auction.feeNumerator()))
```

Comment:
We need to check that this uint256 does not overflow uint96 (`type(uint96).max`)

Reply:
👍